src/api/app/models/user.rb

```diff
@@ -220,7 +220,69 @@ def state=(value)
   # in the database. Returns the user or nil if he could not be found
   def self.find_with_credentials(login, password)
     # Find user
-    user = User.where(login: login).first
+    user = find_by_login(login)
+
+    if CONFIG['ldap_mode'] == :on
+      begin
+        require 'ldap'
+        logger.debug( "Using LDAP to find #{login}" )
+        ldap_info = UserLdapStrategy.find_with_ldap( login, password )
+      rescue LoadError
+        logger.warn "ldap_mode selected but 'ruby-ldap' module not installed."
+        ldap_info = nil # now fall through as if we'd not found a user
+      rescue Exception
+        logger.debug "#{login} not found in LDAP."
+        ldap_info = nil # now fall through as if we'd not found a user
+      end
+
+      return nil unless ldap_info
+
+      # We've found an ldap authenticated user - find or create an OBS userDB entry.
+      if user
+        # Check for ldap updates
+        if user.email != ldap_info[0] or user.realname != ldap_info[1]
+          user.email = ldap_info[0]
+          user.realname = ldap_info[1]
+          user.save
+        end
+        user.login_failure_count = user.login_failure_count + 1
```

Comment:
Maybe I'm missing something... but increasing the login_**failure**_count every time you found the user from ldap is ok? I think that this should reset the failure count not increase it (like at line 228 of this file before you changed it).